internal/provider/resources/deployment.go

```diff
@@ -315,15 +319,13 @@ func (r *DeploymentResource) Schema(_ context.Context, _ resource.SchemaRequest,
 			"concurrency_limit": schema.Int64Attribute{
 				Description: "The deployment's concurrency limit.",
 				Optional:    true,
-				Computed:    true,
```

Comment:
here's another tricky quirk.  So `concurrency_limit` is an optional value for us in the API - but when its set, it's expected to be > 0

When we used to have this as `Computed` and later switched to using req.Plan - same thing, a missing `concurrency_limit` attribute in the HCL results in an `unknown` value

Later, we attach this to the API request like so:

```golang
	createPayload := api.DeploymentCreate{
		ConcurrencyLimit:       plan.ConcurrencyLimit.ValueInt64Pointer(),
        }
```

`ValueInt64Pointer` has this documentation:

> func (i basetypes.Int64Value) ValueInt64Pointer() *int64
ValueInt64Pointer returns a pointer to the known int64 value, nil for a null value, or a pointer to 0 for an unknown value.

Because it was `unknown`, the API payload received a pointer to the value `0` - and that was being sent to the API

instead, we want to have this be `null` when its not set, so that the API call omits the key entirely

Reply:
Thanks, this helps me better understand when to use Computed in general